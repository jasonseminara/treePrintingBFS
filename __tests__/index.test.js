/* eslint-env jest */
const Node = require('../Node');
const printTree = require('../');

const buildTree = (len) => {
  const nodes = Array(len + 1).fill(0).map((n, i) => (new Node(i)));

  nodes[1].children = nodes.slice(2, 5);
  nodes[2].children = nodes.slice(5, 7);
  nodes[4].children = nodes[7];
  nodes[7].children = nodes[9];
  nodes[6].children = nodes[8];
  return nodes[1];
};

describe('TreePrinting', () => {
  let tree;

  beforeEach(() => {
    tree = buildTree(9);
  });

  it('__check tree__', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should print out a tree in level order', () => {

  });
});
