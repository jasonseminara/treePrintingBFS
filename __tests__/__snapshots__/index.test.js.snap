// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreePrinting __check tree__ 1`] = `
Node {
  "kids": Array [
    Node {
      "kids": Array [
        Node {
          "kids": Array [],
          "name": 5,
        },
        Node {
          "kids": Array [
            Node {
              "kids": Array [],
              "name": 8,
            },
          ],
          "name": 6,
        },
      ],
      "name": 2,
    },
    Node {
      "kids": Array [],
      "name": 3,
    },
    Node {
      "kids": Array [
        Node {
          "kids": Array [
            Node {
              "kids": Array [],
              "name": 9,
            },
          ],
          "name": 7,
        },
      ],
      "name": 4,
    },
  ],
  "name": 1,
}
`;
